ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -42,6 +44,21 @@ export default class PermissionConnect extends Component {
     originName: this.props.originName,
   }
 
+  beforeUnload = () => {
+    const { permissionsRequestId, rejectPermissionsRequest } = this.props
+    const { permissionAccepted } = this.state
+
+    if (permissionAccepted === null && permissionsRequestId) {
+      rejectPermissionsRequest(permissionsRequestId)
+    }
+  }
+
+  removeBeforeUnload = () => {
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_FULLSCREEN) {
```

Comment:
When will the environment type not be fullscreen, and why do we not want to reject the request if it isn't?

Reply:
The two other environments are popup and notification. Popup is the one that appears when clicking the browser action button (i.e. the fox in the toolbar), and that one _really_ shouldn't do anything upon close because it only stays open when your mouse is over it.

The notification view is used for confirmation screens currently. It was used for the old provider approval screens as well. This view won't be opened when a permissions request is received, but the user might have a notification window left open from and old or unresolved transaction confirmation or something. If that's the case, they'll end up seeing this permission request screen in _both_ the notification window and the fullscreen tab we opened. I figured we wouldn't want to assume a rejection if the user happens to prune their old windows/tabs while the request is pending.

The fullscreen tab is opened immediately when the request is received, and the user's attention is directed to it. It should be clear I hope that this is the "main" view for this interaction.

Reply:
Solid, thanks for the exhaustive answer.